nil pointer panic when running encrypted sif image

```
[vagrant@localhost test]$ /usr/bin/apptainer version
1.3.6-1.el9

[vagrant@localhost test]$ cat /etc/os-release 
NAME="Rocky Linux"
VERSION="9.5 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.5"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.5 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
VENDOR_NAME="RESF"
VENDOR_URL="https://resf.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.5"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.5"
```

```
[vagrant@localhost test]$ sudo /usr/bin/apptainer build --passphrase alpine_pass_1234.sif docker://alpine:latest
Enter encryption passphrase: 
INFO:    Starting build...
Copying blob f18232174bc9 skipped: already exists  
Copying config aded1e1a5b done   | 
Writing manifest to image destination
2025/03/13 05:29:50  info unpack layer: sha256:f18232174bc91741fdf3da96d85011092101a032a93a388b79e99e69c2d5c870
INFO:    Creating SIF file...
INFO:    Build complete: alpine_pass_1234.sif
```

```
[vagrant@localhost test]$ sudo /usr/bin/apptainer exec --passphrase alpine_pass_1234.sif /bin/true
Enter encryption passphrase: 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x557e33475ccc]

goroutine 3 [running]:
github.com/apptainer/apptainer/internal/pkg/runtime/engine/apptainer.umount()
	github.com/apptainer/apptainer/internal/pkg/runtime/engine/apptainer/cleanup_linux.go:174 +0x12c
github.com/apptainer/apptainer/internal/pkg/runtime/engine/apptainer.cleanupCrypt({0xc000221bf0, 0x30})
	github.com/apptainer/apptainer/internal/pkg/runtime/engine/apptainer/cleanup_linux.go:245 +0x25
github.com/apptainer/apptainer/internal/pkg/runtime/engine/apptainer.(*EngineOperations).CleanupContainer(0xc0001772f0, {0x557e3376e420, 0x557e33c93e40}, {0x557e00000000?, 0x3364f200?}, 0x88008?)
	github.com/apptainer/apptainer/internal/pkg/runtime/engine/apptainer/cleanup_linux.go:136 +0x831
github.com/apptainer/apptainer/internal/app/starter.Master(0x6, 0x4, 0xfc4d8, 0xc000012060)
	github.com/apptainer/apptainer/internal/app/starter/master_linux.go:164 +0x277
main.startup()
	./main_linux.go:84 +0x394
created by main.main in goroutine 1
	./main_linux.go:111 +0x1a
```

it looks like that this line causes this issue
https://github.com/apptainer/apptainer/blob/c8ff5e3fb4e99527e519b90d9917361044cfdd6e/internal/pkg/runtime/engine/apptainer/cleanup_linux.go#L174

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,8 @@ Changes since 1.4.0-rc.2
 
 - Fix `target: no such file or directory` error when extracting
   layers from certain OCI images that manipulate hard links across layers.
+- Fix a nil pointer panic cauesd by the unintialized image driver with
+  privileged encryption.
 
 ## v1.4.0 Release Candidate 2 - \[2025-03-4\]
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,8 +11,8 @@ Changes since 1.4.0-rc.2
 
 - Fix `target: no such file or directory` error when extracting
   layers from certain OCI images that manipulate hard links across layers.
-- Fix a nil pointer panic cauesd by the unintialized image driver with
-  privileged encryption.
+- Fix the crash that happens when executing a privilege-encrypted container as
+  root.  
 
 ## v1.4.0 Release Candidate 2 - \[2025-03-4\]
 

--- a/internal/pkg/runtime/engine/apptainer/cleanup_linux.go
+++ b/internal/pkg/runtime/engine/apptainer/cleanup_linux.go
@@ -171,7 +171,9 @@ func umount() (err error) {
 	}()
 
 	// empty target to signify to driver we are entering in the stop phase
-	imageDriver.Stop("")
+	if imageDriver != nil {
+		imageDriver.Stop("")
+	}
 
 	// gocryptfs related temp folders
 	var gocryptfsTmp []string

